core/src/main/scala/cats/arrow/ArrowChoice.scala

```diff
@@ -0,0 +1,43 @@
+package cats
+package arrow
+
+import simulacrum.typeclass
+
+/**
+ * Must obey the laws defined in cats.laws.ArrowChoiceLaws.
+ */
+@typeclass trait ArrowChoice[F[_, _]] extends Arrow[F] with Choice[F] { self =>
+
+  /**
+   * ArrowChoice yields Arrows with choice, allowing distribution
+   * over coproducts.
+   *
+   * Given two `F`s (`f` and `g`), create a new `F` with
+   * domain the coproduct of the domains of `f` and `g`,
+   * and codomain the coproduct of the codomains of `f` and `g`.
+   * This is the sum notion to `split`'s product.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val toLong: Int => Long = _.toLong
+   * scala> val toDouble: Float => Double = _.toDouble
+   * scala> val f: Either[Int, Float] => Either[Long, Double] = toLong +++ toDouble
+   * scala> f(Left(3))
+   * res0: Either[Long,Double] = Left(3)
+   * scala> f(Right(3))
+   * res1: Either[Long,Double] = Right(3.0)
+   * }}}
+   */
+  @simulacrum.op("+++", alias = true)
+  def choose[A, B, C, D](f: F[A, C])(g: F[B, D]): F[Either[A, B], Either[C, D]]
+
+  def left[A, B, C](fab: F[A, B]): F[Either[A, C], Either[B, C]] =
+    choose(fab)(lift(identity[C]))
+
+  def right[A, B, C](fab: F[A, B]): F[Either[C, A], Either[C, B]] =
+    choose(lift(identity[C]))(fab)
```

Comment:
This is never tested by law as I noticed from the law set defined in Haskell. Thus it doesn't have any test coverage. 
Shall we add a law for this something like 
```scala
  def leftRightConsistent[A, B, C](f: A => B): IsEq[F[Either[A, C], Either[B, C]]] =
     F.right[A, B, C](F.lift[A, B](f)) <-> F.left[A, B, C](F.lift[A, B](f)).dimap(_.swap, _.swap)
```

Reply:
I actually added symmetric versions for all the laws last night for that reason, but it takes the method well over 50 lines. Something like that sounds like a more reasonable way to get similar effect. I'll try it out today, it certainly should work.